client/src/app/site/pages/organization/pages/mediafiles/modules/organization-mediafile-list/components/organization-mediafile-list/organization-mediafile-list.component.ts

```diff
@@ -114,6 +123,42 @@ export class OrganizationMediafileListComponent
         this.cd.detach();
     }
 
+    public isMediafileUsed(file: ViewMediafile, place: LogoPlace): boolean {
+        const mediafile = this.repo.getViewModel(file.id)!;
+        if (file.isImage() && !Object.keys(LogoDisplayNames).includes(place)) {
+            return false;
+        }
+        return mediafile.token === `global-${place}`;
+    }
+
+    public async toggleMediafileUsage(event: Event, mediafile: ViewMediafile, place: LogoPlace): Promise<void> {
+        // prohibits automatic closing
+        event.stopPropagation();
+        this.updatingLogoAndFontSettings = true;
+        const file = this.repo.getViewModel(mediafile.id);
+        if (!file || (file.isImage() && !Object.keys(LogoDisplayNames).includes(place))) {
+            throw new Error(!file ? `File has been deleted` : `Invalid mediafile type for place.`);
+        }
+        const fullPlace = `global-${place}`;
+        if (!(file.token === fullPlace)) {
+            for (let filteredFile of this.repo
+                .getViewModelList()
+                .filter(filterFile => filterFile.token === fullPlace && filterFile.id !== file.id)) {
+                await this.repo.update({ token: null }, filteredFile);
+            }
+            await this.repo.update({ token: fullPlace }, file);
```

Comment:
This should be done in one transaction in the backend. I will open an issue for this. For now, this can stay as-is.